### PR TITLE
[ENH] `names_expand` and `index_expand`

### LINKED ISSUE #1108: suggestion - Add `names_expand` to `pivot_wider`
# Brief Description

I would like to propose adding a `names_expand` parameter to `pivot_wider` - this expands the columns to all possible combinations (useful for categorical columns).  ``df.pivot_wider(index, names_from, values_from, names_expand = True/False)``

# Example API

```python
# adapted from https://www.tidyverse.org/blog/2022/02/tidyr-1-2-0/
import pandas as pd
weekdays = ("Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun")
daily = {'day': pd.Categorical(("Tue", "Thu", "Fri", "Mon"), categories = weekdays),
              'value' : (2,3,1,5)}

daily = pd.DataFrame(daily, index = [0,0,0,0])

daily
   day  value
0  Tue      2
0  Thu      3
0  Fri      1
0  Mon      5

 daily.dtypes
day      category
value       int64
dtype: object

 daily.day
0    Tue
0    Thu
0    Fri
0    Mon
Name: day, dtype: category
Categories (7, object): ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']


# pivot only shows the existing values from the `day` column
daily.pivot_wider(index = None, names_from = 'day', values_from = 'value')
   Tue  Thu  Fri  Mon
0    2    3    1    5

# one efficient option is with reindexing on the columns
# which is what `names_expand` will use under the hood
daily.pivot_wider(index = None, names_from = 'day', values_from = 'value').reindex(columns = weekdays)

   Mon  Tue  Wed  Thu  Fri  Sat  Sun
0    5    2  NaN    3    1  NaN  NaN

# note however that we had to grab the weekdays to do this
# instead of using the categories within `names_from`
# if there were other columns, then we'd have to set_index, reindex, then reset_index (delaying the reset_index - doing the operations once before resetting would be ideal)
# and if there were multiple `names_from`, reindexing on one level won't suffice.
```

### PR DESCRIPTION
**Pull request by samukweku**
Please describe the changes proposed in the pull request:

- Builds on similar idea from tidyr `pivot_wider`
- Allow exposure of all categoricals either in the columns or index
- preserve categories where possible

**This PR resolves #1108**

# PR Checklist

Please ensure that you have done the following:

1. [ ] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.md`.

3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Comment by ericmjl** [comment]
🚀 Deployed on https://deploy-preview-1126--pyjanitor.netlify.app

**Comment by samukweku** [comment]
test failures will be fixed after review of #1122

**Comment by ericmjl** [comment]
@samukweku let's get the merge conflicts resolved please! In the meantime, I'll take a look at the diff.

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,7 +10,7 @@
 -   [ENH] `min_max_scale` drop `old_min` and `old_max` to fit sklearn's method API. Issue #1068 @Zeroto521
 -   [ENH] Add `jointly` option for `min_max_scale` support to transform each column values or entire values. Default transform each column, similar behavior to `sklearn.preprocessing.MinMaxScaler`. (Issue #1067, PR #1112, PR #1123) @Zeroto521
 -   [INF] Require pyspark minimal version is v3.2.0 to cut duplicates codes. Issue #1110 @Zeroto521
-
+-   [ENH] Add names_expand and index_expand parameters to pivot_wider for exposing missing categoricals. Issue #1108 @samukweku
```
Comment (ericmjl) [inline CHANGELOG.md]: Minor nit:

```suggestion
-   [ENH] Add `names_expand` and `index_expand` parameters to `pivot_wider` for exposing missing categoricals. Issue #1108 @samukweku
```

**Review by ericmjl — CHANGES_REQUESTED** [review]
Looked through the PR; LGTM, @samukweku. The only thing I'm a bit concerned about is codecov. Do you have an idea why the coverage dropped so much?

File: CHANGELOG.md (hunk 1 above)
Comment (samukweku) [reply CHANGELOG.md]: It was dependent on other PRs. I'll update and hopefully codecov coverage should be better

**Comment by samukweku** [comment]
@ericmjl resolved the codecov coverage drop to an acceptable level hopefully

**Comment by ericmjl** [comment]
Looks good to me. Let's merge!

**Pull request merged**